README.md

```diff
@@ -149,6 +149,7 @@ the discretion of the Release working group.
 [Erbium]: https://nodejs.org/download/release/latest-erbium/
 [Fermium]: https://nodejs.org/download/release/latest-fermium/
 [Gallium]: https://nodejs.org/download/release/latest-gallium/
+[Hydrogen]: https://nodejs.org/download/release/latest-hydrogen/
```

Comment:
This needs a PR in https://github.com/nodejs/nodejs-latest-linker/ to add the mapping to create this link. If we haven't documented that as a step in the release process, we'll need to add it for the LTS transition release.